### PR TITLE
Add shell completion

### LINKED ISSUE #39: [FEATURE REQ] Tab completion for grofer.
**Is your feature request related to a problem? Please describe.**
The grofer command would be easier to use with tab completion!

**Describe the solution you'd like**
As we're implementing cobra to parse command line options and arguments, we could enhance it with tab auto completions as cobra supports this for major shells.

### PR DESCRIPTION
**Pull request by anihm136**
# Description

Adds a completion subcommand to grofer, which adds the ability to generate completion scripts for bash, zsh, fish and powershell

Fixes #39

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Checklist:

- [X] I have read the [contribution guidelines](https://github.com/pesos/grofer/blob/master/CONTRIBUTING.md) and followed it as far as possible. 
- [X] I have performed a self-review of my own code (if applicable)
- [ ] I have commented my code, particularly in hard-to-understand areas (if applicable)
- [X] I have run `go fmt` on my code ([reference](https://blog.golang.org/gofmt))
- [ ] I have made corresponding changes to the documentation (if applicable)
- [X] My changes generate no new warnings 
- [ ] Any dependent and pending changes have been merged and published

### CODE REVIEW

**Review by Gituser143 — CHANGES_REQUESTED** [review]
@anihm136 You're missing the license on the new files. You'll have to put a header at the beginning of the file. You can copy the license from an existing .go file.

**Comment by anihm136** [comment]
Done

**Comment by anihm136** [comment]
Looks like its still failing on the license check?

**Comment by Gituser143** [comment]
> Looks like its still failing on the license check?

For some reason it's still building the previous commit. I'll look into this.

**Review by Gituser143 — APPROVED** [review]

**Comment by Gituser143** [comment]
@anihm136 The license check is failing because you seem to have extra white space on line number 8. I'd suggest you copy the license straight from the license checking script in `scripts/license_check.py` to avoid this weird issue.

File: cmd/completion.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/*
+Copyright © 2020 The PES Open Source Team pesos@pes.edu
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
```
Comment (Gituser143) [inline cmd/completion.go:8]: This is where you've got extra white space.

**Review by Gituser143 — CHANGES_REQUESTED** [review]

File: cmd/completion.go (hunk 1 above)
Comment (anihm136) [reply cmd/completion.go:8]: Finally caught it. Tabs vs whitespace

File: cmd/completion.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Zsh:
+
+# If shell completion is not already enabled in your environment you will need
+# to enable it.  You can execute the following once:
+
+$ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+# To load completions for each session, execute once:
+$ yourprogram completion zsh > "${fpath[1]}/_grofer"
```
Comment (Gituser143) [inline cmd/completion.go:49]: @anihm136 Here's an additional thing you might want to change. I'm assuming `yourprogram` should be `grofer`?

Additionally, this requires documentation updates, so perhaps start with including a section in the README for completions?

**Review by Gituser143 — CHANGES_REQUESTED** [review]

File: cmd/completion.go (hunk 2 above)
Comment (anihm136) [reply cmd/completion.go:49]: Yeah, looks like I missed that one. I'll add a section

**Comment by anihm136** [comment]
Done

**Review by Gituser143 — APPROVED** [review]

**Comment by Gituser143** [comment]
@anihm136 This is perfect! I'll wait for @MadhavJivrajani or @metonymic-smokey to approve and merge.

**Review by MadhavJivrajani — APPROVED** [review]
Looks great @anihm136! Thanks!

**Pull request merged**